### ISSUE TITLE
Ditch `pickle` in favor of our custom `scriber`

### ISSUE THREAD

**Issue by supersergiy** [seung-lab/corgie#16]
Many benefits. After this is done, revisit the approach used for passing the device to Render.

**Comment by william-silversmith** [comment]
Not sure what scriber does, but you may possibly be interested in https://github.com/seung-lab/mapbuffer.

**Comment by supersergiy** [comment]
Scriber doesn't do anything yet, I need to make it :D

What I'm looking for is basically the kind of serialization that's done for tasks in `igneous`, but I also need it to be recursive, aka supporting tasks that have other tasks as attributes.

**Comment by william-silversmith** [comment]
This is starting to get into seuron territory with task dependencies. I think we discussed why that wasn't a good fit a while ago, but can you remind me? Being able to do everything using only a filesystem queue is nice.

Do you think scriber's functionality would fit in TaskQueue?

**Comment by supersergiy** [comment]
Yeah, I remember we discussed this a lot right before I left for Samsung. Basically we don't want to switch to seuron just yet because it's nice to be able to write code imperatively instead of building a dependency graph upfront. Right now we're able to write our alignment job as:
```
for z in range(z_start, z_end):
            compute_field_job = self.cf_method(......)

            yield from compute_field_job.task_generator
            yield scheduling.wait_until_done

            render_job = self.render_method(.......)

            yield from render_job.task_generator
            yield scheduling.wait_until_done
```
which I think reads really nicely and is intuitive to write and debug.  This method is not as expressive as seuron -- we can only run series of independent jobs, with Barrier type dependencies within each job. So far the barriers have been enough to satisfy the alignment needs. We might switch to seuron backend later on, but not unless we have to.

**Comment by supersergiy** [comment]
At the same time, there are cases where we want nested tasks, mainly for code reuse. I started writing some code for recursive serialization back in the day, but gave it up because I thought pickle might be good enough. Turns out it's really not. 

It would be great if TaskQueue supports nested task classes too! After all, our whole scheduling system is just a layer on top of TaskQueue

**Comment by william-silversmith** [comment]
How are the tasks nested? Are they just daisy chained or something more
sophisticated?

**Comment by supersergiy** [comment]
Nothing crazy, chaining would probably cover most cases, but ideally I'd like to keep it as flexible as possible.

**Comment by william-silversmith** [comment]
I guess what you want then is to be able to specify a tree/forest of
execution since you aren't looking to be able to specify execution in the
middle of another task.
